crates/shared/src/transport/extensions.rs

```diff
@@ -0,0 +1,143 @@
+//! Module containing Ethereum RPC extension methods.
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use web3::{
+    self,
+    api::Namespace,
+    helpers::{self, CallFuture},
+    types::{BlockId, Bytes, CallRequest, H160, H256, U256, U64},
+    Transport,
+};
+
+/// Ethereum RPC extension methods that are not part of the JSON RPC standard
+/// but commonly implemented by nodes.
+#[derive(Debug, Clone)]
+pub struct EthExt<T> {
+    transport: T,
+}
```

Comment:
Is this struct needed?
Couldn't it just be a trait extension for every type which implements `Transport`?

Reply:
Yes... it can. I actually didn't think much about it and just followed the pattern that the `web3` crate uses [with their other APIs](https://docs.rs/web3/latest/src/web3/api/mod.rs.html#39-46).

I actually don't have too many strong feeling here. I think the idea is that you could do things like:

```rust
let eth_ext = web3.eth_ext();
// ...
eth_ext.call(...).await?;
```

But, I agree that adding it as an extension trait to the existing `web3::api::Eth` works nicely:
```rust
web3.eth().call_ext(...).await?;
```

I'm pretty indifferent either way - let me know if you prefer one of the other.

Reply:
I forgot that we couldn't simply overload the `call()` function but I think a trait extension with `call_with_overrides()` (or sth like that) would be a bit nicer and more versatile (fewer changes when some component suddenly wants to use overrides).

Reply:
I somewhat prefer not introducing a new struct.

Reply:
Refactored into an extension trait.